pkg/server/server.go

```diff
@@ -354,13 +353,16 @@ func (s *Server) HTTPHandler() http.Handler {
 		panic(err)
 	}
 	opts := []graphql.SchemaOpt{graphql.UseFieldResolvers()}
-	k8sResolver := resolver.K8sResolver{K8sProxy: localK8sProxy}
-	rootResolver := resolver.RootResolver{K8sResolver: &k8sResolver}
-	schema := graphql.MustParseSchema(string(graphQLSchema), &rootResolver, opts...)
-	handler := graphqlws.NewHandler()
-	handler.InitPayload = resolver.InitPayload
-	graphQLHandler := handler.NewHandlerFunc(schema, &relay.Handler{Schema: schema})
-	handle("/api/graphql", authHandlerWithUser(func(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	handle(graphQLEndpoint, authHandlerWithUser(func(user *auth.User, w http.ResponseWriter, r *http.Request) {
+		cluster := serverutils.GetCluster(r)
+		k8sProxyConfig := s.getK8sProxyConfig(cluster)
+		k8sProxy := proxy.NewProxy(k8sProxyConfig)
+		k8sResolver := resolver.K8sResolver{K8sProxy: k8sProxy}
+		rootResolver := resolver.RootResolver{K8sResolver: &k8sResolver}
+		schema := graphql.MustParseSchema(string(graphQLSchema), &rootResolver, opts...)
+		handler := graphqlws.NewHandler()
+		handler.InitPayload = resolver.InitPayload
+		graphQLHandler := handler.NewHandlerFunc(schema, &relay.Handler{Schema: schema})
```

Comment:
@TheRealJon you already had the backend part implemented :)

Reply:
I'm about to change this implementation a little, but it should be an easy adjustment. Instead of creating a new proxy instance on every request, we should create one on the first request, then reuse the same proxy instance on subsequent requests.

Reply:
that is happening already IIUC. The handle function is executed once, when graphl client tries to connect to graphql server - we create a proxy and graphql handler here. The rest of the communication is done via websocket and the graphql handler is always using the same proxy instance.
When the cluster is changed in the UI, the websocket is closed and new one is created, executing the handle function again (with new cluster as parameter)

Reply:
Yes, but I think the number of proxy instances still scales up with the number of requests, which is probably not a scalable solution. With many users/sessions, proxy instances increase.